File: 137.SingleNumberII/sort.js
var singleNumber = function (nums) {
  nums.sort();
  for (let i = 0; i < nums.length - 1; i += 3) {
    if (nums[i] == nums[i + 1]) {
      continue;
    } else {
      return nums[i];
    }
  }
  return nums[nums.length - 1];
};

File: 1365.HowManyNumbersAreSmallerThantheCurrentNumber/Sort.java
import java.util.*;

class Sort {
  public int[] smallerNumbersThanCurrent(int[] nums) {
    int n = nums.length;
    int[][] arr = new int[n][2];

    for (int i = 0; i < n; i++) {
      arr[i][0] = nums[i];
      arr[i][1] = i;
    }

    Arrays.sort(arr, (a, b) -> Integer.compare(a[0], b[0]));

    int[] ans = new int[n];
    int prevCount = 0;

    for (int i = 0; i < n; i++) {
      if (i == 0 || arr[i][0] != arr[i - 1][0]) {
        prevCount = i;
      }
      ans[arr[i][1]] = prevCount;
    }

    return ans;
  }
}
